src/iexamine.cpp

```diff
@@ -116,164 +111,135 @@ void iexamine::atm(player *p, map *m, int examx, int examy)
 
     amenu.addentry( cancel, true, 'q', _("Cancel") );
     amenu.query();
-    choice = amenu.ret;
+    auto const choice = amenu.ret;
     uistate.iexamine_atm_selected = choice;
 
-    if (choice == deposit_money) {
-        pos = g->inv(_("Insert card for deposit."));
-        dep = &(p->i_at(pos));
+    //----------------------------------------------------------------------------------------------
+    // Small helper for selecting a card from the inventory.
+    auto const choose_card = [&](char const *const msg) -> item* {
+        auto const index = g->inv_for_filter(msg, [](item const& itm) {
+            return itm.type->id == "cash_card";
+        });
 
-        if (dep->is_null()) {
-            popup(_("You do not have that item!"));
-            return;
+        if (index == INT_MIN) {
+            return nullptr; // player canceled
         }
-        if (dep->type->id != "cash_card") {
+
+        auto &itm = p->i_at(index);
+        if (itm.type->id != "cash_card") {
             popup(_("Please insert cash cards only!"));
+            return nullptr; // must have selected an equipped item
+        }
+
+        return &itm;
+    };
+
+    //----------------------------------------------------------------------------------------------
+    // Small helper to promp for an amount and clamp it to [0, max].
```

Comment:
That sounds like a thing that would be useful in other places, but if it's a lambda you can only use it here.